Neos.Fusion/Classes/Core/ObjectTreeParser/FilePatternResolver.php

```diff
@@ -131,6 +131,10 @@ protected static function iterateOverFilesAndSelectByFileEnding(\Iterator $fileI
     {
         $files = [];
         foreach ($fileIterator as $fileInfo) {
+            // avoid hitting possible upwards traversal protection
```

Comment:
hmmm

Reply:
this seems to protect against this correct?

`include ../../../NodeTypes/**/*.fusion`

we should add a testcase: see https://github.com/neos/neos-development-collection/blob/8.2/Neos.Fusion/Tests/Unit/Core/Parser/ParserIncludeTest.php

Reply:
The protection is tested in "openThrowsExceptionForPathesThatTryToTraverseUpwards" in the NodeTyppesStreamWrapperTest

Reply:
i know - but thats a Neos.Neos test for the stream wrapper itself.

here you modifed Neos.Fusion `ObjectTreeParser/FilePatternResolver.php` and i cannot really imagine why ^^

Reply:
Because it will call is_dir on NodeTypes/.. an this will cause an exception as it is out of the allowed scope

Reply:
Or more concrete the directory list of the root paths contains `..` and the FilePatternResolver will check for every thing wether it is a directory. This is detected as a scope violation and causes an error.

The alternative would be to not return `..` in the root directory of a `nodetypes://` path but that would require a lot of additional state in the stream wrapper because `readDirectory` has no additional arguments that would allow to detect that it is traversing a root path in nodetypes.

Reply:
@mhsdesign found out the flag \FilesystemIterator::SKIP_DOTS on the recursive fileysystem interator achieves the same with much less weirdness.

Reply:
nice thank you!

i like the solution mich better and also now better understand ^^

should we add this also to: return new \DirectoryIterator($dir);?

Reply:
Has to revert that because of the normal DirectoryIterator not having those flags ... now i filter out the dotfiles in the "iterateOverFilesAndSelectByFileEnding" method again ... gnarf

Reply:
sorry for the extra work :O 

but lets merge ;) - the workaround is fine

Reply:
And adjusted again .. found out that the "FilesystemIterator" supports the SkipDots flag ... hope all tests will run green with this now.